clang/lib/Sema/SPIRVBuiltins.td

```diff
@@ -267,34 +271,118 @@ class Builtin<string _Name, list<Type> _Signature, list<bit> _Attributes = Attr.
 
 // Helper to declare SPIR-V Core builtins.
 class SPVBuiltin<string _Name, list<Type> _Signature, list<bit> _Attributes = Attr.None> :
-Builtin<"__spirv_" # _Name, _Signature, _Attributes> {} 
+  Builtin<"__spirv_" # _Name, _Signature, _Attributes> {}
 
 // Helper to declare OpenCL SPIR-V extended set builtins.
 class OCLSPVBuiltin<string _Name, list<Type> _Signature, list<bit> _Attributes = Attr.None> :
-SPVBuiltin<"ocl_" # _Name, _Signature, _Attributes> {} 
+  SPVBuiltin<"ocl_" # _Name, _Signature, _Attributes> {}
+
+class ConstOCLSPVBuiltin<string _Name, list<Type> _Signature> :
+  OCLSPVBuiltin<_Name, _Signature, Attr.Const> {}
 
 //===----------------------------------------------------------------------===//
 //                 Definitions of types
 //===----------------------------------------------------------------------===//
 
+// OpenCL v1.0/1.2/2.0 s6.1.1: Built-in Scalar Data Types.
+def Bool      : IntType<"bool",   QualType<"BoolTy">, 1>;
```

Comment:
BTW @bader I added `Bool` to keep thing as they are ATM in the SYCL runtime. Although, as per mentioned in this PR https://github.com/intel/llvm/pull/1287, I think lowering to boolean should be handled by the translator and the builtin definition should accept `char` or `uchar` and vector of them. One of the reason is that not all languages have the type bool.

Reply:
Scalar bool must be supported.
The only problem is vector bool.